### ISSUE TITLE
Replaceable class types are not shown in the parameter dialog with the new API

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#10140]
### Description

Take a class that includes a replaceable type definition with choices. If you instantiate this class, the dropdown menu for the replaceable type choices does not appear in the parameter dialog with the new API. With the old API this works fine. The new API is only able to display the dropdown menu for replaceable instances.

### Steps to Reproduce

1. Create a class with a replaceable type definition like (or take Rexroth_ElectricDrives.RexrothRotationalDrives.SynchronousRotationalDrives.KSM02.KSM02 from the BRSL Rexroth_ElectricDrives library): 
```modelica
replaceable record Motor = Rexroth_ElectricDrives.Internal.Database.KSM02.KSM02p1B_041C_42N_xx_xx0_xx_xx_xx_xxxxx "Motor" annotation (Dialog(group="Configuration parameters"));
```
2. Create a new class and instantiate the class from step 1. Open its parameter dialog. The replaceable record `Motor` is not shown in the parameter dialog as depicted in the first screenshot. If you open the same model with the old API the replaceable record is shown (see second screenshot).

### Expected Behavior

Replaceable types are shown in the parameter dialog with the new API just like with the old one.

### Screenshots

Will be added by @niklwors because I am not allowed to upload them from my work laptop.

### Version and OS

 - OpenModelica Version: OMEdit 1.21-dev
 - OS: Windows 10, 64 bit

Thanks for looking into it:
Aaron Buntrock - Bosch rexroth

**Comment by niklwors** [comment]
Picture 1:
![image003](https://user-images.githubusercontent.com/12269776/216351103-6eb1e070-034f-44a6-b2de-436e3f25af0b.png)

Picture 2:
![image004](https://user-images.githubusercontent.com/12269776/216351123-6ddf2957-88c5-4d52-8025-d8a8bf36504b.png)

**Comment by abuntrock** [comment]
When looking into this, maybe its possible to fix issue #10030 for the new API in one go as well.

**Comment by adeas31** [comment]
The actual issue reported in this ticket is fixed. The modification of the redeclared class is not perfect but we are working on it.
